libethereum/Executive.h

```diff
@@ -150,7 +151,7 @@ class Executive
 
 	/// Set up the executive for evaluating a bare CREATE (contract-creation) operation.
 	/// @returns false iff go() must be called (and thus a VM execution in required).
-	bool create(Address _txSender, u256 _endowment, u256 _gasPrice, u256 _gas, bytesConstRef _code, Address _originAddress);
+	bool create(Address _txSender, u256 _endowment, u256 _gasPrice, u256 _gas, bytesConstRef _code, Address _originAddress, CreationContext _creationType);
```

Comment:
I would actually do this not with an `enum` parameter, but by having three separate interface methods, something like `create`, `createOpcode`, `createP2shOpcode`. Each would calculate new address in its own way and then call the internal implementation function with the rest of the code.

Branching he logic on enum is more error-prone and less maintainable, because if we add a new enum value, we need to search for all the places where we branch on this enum. We'd also get a nicer smaller functions and less nested `if` constructions, if we extract the address calculation from the current big `create`

Sorry for not pointing this out earlier, it's probably not that critical and you could leave it like this if you think it's better